gcore/gdaldrivermanager.cpp

```diff
@@ -1146,6 +1219,496 @@ void GDALDriverManager::ReorderDrivers()
+        if (m_poRealDriver->pfnIdentify)
+            pfnIdentify = m_poRealDriver->pfnIdentify;
+
+        CheckFunctionPointer(
+            reinterpret_cast<void *>(m_poRealDriver->pfnGetSubdatasetInfoFunc),
+            reinterpret_cast<void *>(pfnGetSubdatasetInfoFunc),
+            "pfnGetSubdatasetInfoFunc");
+
+        const auto CheckFunctionPointerVersusCap =
+            [this](void *pfnFunc, const char *pszFunc, const char *pszItemName)
+        {
+            if (pfnFunc && !GetMetadataItem(pszItemName))
+            {
+                CPLError(CE_Warning, CPLE_AppDefined,
+                         "Driver %s declares a %s callback whereas its proxy "
+                         "doest not declare %s",
+                         GetDescription(), pszFunc, pszItemName);
+            }
+            else if (!pfnFunc && GetMetadataItem(pszItemName))
+            {
+                CPLError(CE_Warning, CPLE_AppDefined,
+                         "Proxy driver %s declares %s whereas the real "
+                         "driver does not declare a %s callback",
+                         GetDescription(), pszItemName, pszFunc);
+            }
+        };
+
+        CheckFunctionPointerVersusCap(
+            reinterpret_cast<void *>(m_poRealDriver->pfnOpen), "pfnOpen",
+            GDAL_DCAP_OPEN);
+        CheckFunctionPointerVersusCap(
+            reinterpret_cast<void *>(m_poRealDriver->pfnCreate), "pfnCreate",
+            GDAL_DCAP_CREATE);
+        CheckFunctionPointerVersusCap(
+            reinterpret_cast<void *>(m_poRealDriver->pfnCreateCopy),
+            "pfnCreateCopy", GDAL_DCAP_CREATECOPY);
+        CheckFunctionPointerVersusCap(
+            reinterpret_cast<void *>(m_poRealDriver->pfnCreateMultiDimensional),
+            "pfnCreateMultiDimensional", GDAL_DCAP_CREATE_MULTIDIMENSIONAL);
+    }
+
+    return m_poRealDriver.get();
+}
+
+/************************************************************************/
+/*                        GetPluginFullPath()                           */
+/************************************************************************/
+
+std::string GDALDriverManager::GetPluginFullPath(const char *pszFilename) const
+{
+    if (!m_osLastTriedDirectory.empty())
+    {
+        const char *pszFullFilename = CPLFormFilename(
+            m_osLastTriedDirectory.c_str(), pszFilename, nullptr);
+        VSIStatBufL sStatBuf;
+        if (VSIStatL(pszFullFilename, &sStatBuf) == 0)
+        {
+            return pszFullFilename;
+        }
+    }
+
+    const char *pszGDAL_DRIVER_PATH =
+        CPLGetConfigOption("GDAL_DRIVER_PATH", nullptr);
+    if (pszGDAL_DRIVER_PATH == nullptr)
+        pszGDAL_DRIVER_PATH = CPLGetConfigOption("OGR_DRIVER_PATH", nullptr);
+
+    /* ---------------------------------------------------------------- */
+    /*      Allow applications to completely disable this search by     */
+    /*      setting the driver path to the special string "disable".    */
```

Comment:
I didn't check but I assume this has been added to the documentation.

Reply:
yes, this is documented in https://gdal.org/api/gdaldriver_cpp.html#_CPPv4N17GDALDriverManager15AutoLoadDriversEv . But this was missing in https://gdal.org/user/configoptions.html#driver-management. Added there too